[PR]: Search-Function/Fetch-Improvements

Sarveshk76 opened the pull request:
1. Sometimes, for unknown reasons, even when language is chosen as a search parameter, the API returns results with different language codes. For instance, author-searching for "Andrzej Sapkowski" in English (en) returned one volume in Italian (it). Would be great to have this solved. **(Can you please share which volume it returns in italian? I didn't find any italian volumes)**
 2. I have iterated over author search to make it more inclusive, but it still needs work. For example, "Emma Mieko Candon" returns appropriate results, as well as "Emma Mieko", but "Emma Candon" returns nothing. **(I have resolved this issue. Please review the code)**
 **And I also updated the useFetchBooks hook**
#9

Sarveshk76 committed:
updated useFetchBooks hook

livoszlak commented:
> 1. Sometimes, for unknown reasons, even when language is chosen as a search parameter, the API returns results with different language codes. For instance, author-searching for "Andrzej Sapkowski" in English (en) returned one volume in Italian (it). Would be great to have this solved. **(Can you please share which volume it returns in italian? I didn't find any italian volumes)**

Sure! It's [this volume](https://books.google.se/books/about/La_stagione_delle_tempeste_The_Witcher.html?hl=&id=zUeOzQEACAAJ&redir_esc=y), which appears when searching for author "Andrzej Sapkowski" with English as selected language. Second page of search results in the pagination (when sorted by relevance).

> 2. I have iterated over author search to make it more inclusive, but it still needs work. For example, "Emma Mieko Candon" returns appropriate results, as well as "Emma Mieko", but "Emma Candon" returns nothing. **(I have resolved this issue. Please review the code)**
>    **And I also updated the useFetchBooks hook**
>    [Search Function/Fetch Improvements #9](https://github.com/livoszlak/book-finder-api-2.0/issues/9)

Great work! Taking a look at your PRs now. 👍

livoszlak commented:
@Sarveshk76 - there is still an issue with searching for author's whose names include initials. For instance, "J R R Tolkien" returns 0 results; "J.R.R. Tolkien" returns results, but not all that would be expected. "R F Kuang" and "R. F. Kuang" return nothing, but "R.F. Kuang" returns expected results. Is there anything else that can be done with regex to solve that issue when it comes to punctuation and matching, do you think? 🤔 

Also, I tried searching for "J.K. Rowling" just to see if the initials problem was true there also, and got some results in Urdu and Spanish. I haven't checked yet, but maybe those results are volumes where no language is found in the response? Just a theory.

Sarveshk76 committed:
Updated the code

Sarveshk76 commented:
Hi @livoszlak, updated the code with issue you mentioned. For the languege filter, google api is not filtering properly using param "langRestrict", so I added a manual filter while rendering the book card.
